webhook/csv.go

```diff
@@ -71,19 +74,33 @@ func (r *ClusterServiceVersionDeploymentScaler) Handle(ctx context.Context, req
 		return admission.Allowed("csv is not managed by ODF")
 	}
 
-	running, err := r.isPreviousCsvHasRunningDeployments(ctx, logger, csv)
-	if err != nil {
-		logger.Error(err, "failed getting replicas from csv")
-		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed getting replicas from csv: %v", err))
+	var noPrevCsvFound bool
+	if csv.Spec.Replaces != "" {
+		prevCsv := &opv1a1.ClusterServiceVersion{}
+		key := client.ObjectKey{Name: csv.Spec.Replaces, Namespace: csv.Namespace}
+		if err := r.Client.Get(ctx, key, prevCsv); err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				logger.Error(err, "failed to get previous CSV", "csv", csv.Spec.Replaces)
+				return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed getting previous csv %q: %w", csv.Spec.Replaces, err))
+			}
+			noPrevCsvFound = true
+		} else {
+			logger.Info("previous CSV found", "csv", csv.Spec.Replaces)
+			defaultsMap := map[string]int32{}
+			if r.isCsvHasRunningDeployments(csv) {
+				defaultsMap[keyReplicas] = 1
+			} else {
+				defaultsMap[keyReplicas] = 0
+			}
+			r.syncNewCsvWithPrevCsv(prevCsv, csv, defaultsMap)
+		}
 	}
 
-	if running {
-		logger.Info("ignoring csv as the previous csv deployments are running")
-		return admission.Allowed("previous csv deployments are running")
+	if noPrevCsvFound {
```

Comment:
This logic is also broken as in the case of a new install, `noPrevCsvFound` will be false (zero value) and it will never scale down. This will be automatically solved if you use the above variable and flip the conditions.